Objects/descrobject.c

```diff
@@ -1791,6 +1794,19 @@ property_init_impl(propertyobject *self, PyObject *fget, PyObject *fset,
         if (rc <= 0) {
             return rc;
         }
+        if (!Py_IS_TYPE(self, &PyProperty_Type) &&
+            prop_doc != NULL && prop_doc != Py_None) {
+            // This oddity preserves the long existing behavior of surfacing
+            // an AttributeError when using a dict-less (__slots__) property
+            // subclass as a decorator on a getter method with a docstring.
+            // See PropertySubclassTest.test_slots_docstring_copy_exception.
+            int err = PyObject_SetAttr(
+                        (PyObject *)self, &_Py_ID(__doc__), prop_doc);
+            Py_DECREF(prop_doc);
```

Comment:
I don't think we need to DECREF here.
```suggestion
```

Reply:
That was indeed the asan bug and crasher. This DECREF belonged in the err < 0 if before the return. _PyObject_LookupAttr had returned a new reference to us in prop_doc at this point.

Reply:
You're right. I think we're also missing a DECREF on the early return on line 1795.

Reply:
(Testing locally to confirm this)

Reply:
No sorry, that's wrong. There's no `propdoc` yet at that point.